hooks/emitter/bandtss.go

```diff
@@ -0,0 +1,232 @@
+	groupIDs := evMap[types.EventTypeInactiveStatus+"."+types.AttributeKeyGroupID]
+	if len(addresses) != len(groupIDs) {
+		panic("invalid event data")
+	}
+
+	for i, addr := range addresses {
+		acc, err := sdk.AccAddressFromBech32(addr)
+		if err != nil {
+			panic(err)
+		}
+
+		groupID := tss.GroupID(common.Atoi(groupIDs[i]))
+		h.handleBandtssUpdateMember(ctx, acc, groupID)
+	}
+}
+
+// handleBandtssEventGroupTransition implements emitter handler for group transition event.
+func (h *Hook) handleBandtssEventGroupTransition(ctx sdk.Context, eventIdx int, querier *EventQuerier) {
+	// if transition not found, skip the process. There is a case that the transition message is signed
+	// at the same block as the transition can be executed. The transition status will be updated via
+	// another event (group_transition_success, group_transition_failed).
+	transition, found := h.bandtssKeeper.GetGroupTransition(ctx)
+	if !found {
+		return
+	}
+
+	// set new bandtss members; better approach would be emitting addMember event.
+	if transition.Status == types.TRANSITION_STATUS_WAITING_EXECUTION {
+		tssMembers := h.tssKeeper.MustGetMembers(ctx, transition.IncomingGroupID)
+		for _, tssMember := range tssMembers {
+			addr := sdk.MustAccAddressFromBech32(tssMember.Address)
+			member, err := h.bandtssKeeper.GetMember(ctx, addr, transition.IncomingGroupID)
+			if err != nil {
+				panic(err)
+			}
+			h.emitSetBandtssMember(member)
+		}
+	}
+
+	// check if it is a new transition or update the status of the existing transition.
+	isNewTransition := transition.Status == types.TRANSITION_STATUS_CREATING_GROUP ||
+		(transition.IsForceTransition && transition.Status == types.TRANSITION_STATUS_WAITING_EXECUTION)
+
+	if isNewTransition {
+		proposalID, found := getCurrentProposalID(eventIdx, querier)
+		if !found {
+			panic("proposal ID not found")
+		}
+
+		h.emitSetBandtssGroupTransition(proposalID, transition, ctx.BlockHeight())
+	} else {
+		h.emitUpdateBandtssGroupTransitionStatus(transition)
+	}
+}
+
+// handleBandtssEventGroupTransitionSuccess implements emitter handler for group transition success event.
+func (h *Hook) handleBandtssEventGroupTransitionSuccess(ctx sdk.Context, evMap common.EvMap) {
+	// use value from emitted event due to the transition info is removed from the store.
+	signingIDs := evMap[types.EventTypeGroupTransitionSuccess+"."+tsstypes.AttributeKeySigningID]
+	incomingGroupIDs := evMap[types.EventTypeGroupTransitionSuccess+"."+types.AttributeKeyIncomingGroupID]
+	currentGroupIDs := evMap[types.EventTypeGroupTransitionSuccess+"."+types.AttributeKeyCurrentGroupID]
+
+	signingID := tss.SigningID(common.Atoi(signingIDs[0]))
+	currentGroupID := tss.GroupID(common.Atoi(currentGroupIDs[0]))
+	incomingGroupID := tss.GroupID(common.Atoi(incomingGroupIDs[0]))
+
+	h.emitUpdateBandtssGroupTransitionStatusSuccess(types.GroupTransition{
+		SigningID:       signingID,
+		CurrentGroupID:  currentGroupID,
+		IncomingGroupID: incomingGroupID,
+	})
```

Comment:
reorder the params which one should be coming frist currentGroupID or incomingGroupID